File: ztm-spa/src/app/components/route/route.service.js
(function() {
  'use strict';

  angular
      .module('ztmSpa')
      .service('RouteService', routeService);

  /** @ngInject */
  function routeService(Route) {

    this.getById = getById;

    function getById(routeId) {
      return Route.getById({routeId: routeId});
    }
  }

})();

File: ztm-services/src/main/java/com/opengdansk/route/service/RouteService.java
package com.opengdansk.route.service;

import com.opengdansk.route.configuration.RouteMapBean;
import com.opengdansk.route.model.Route;
import lombok.NonNull;
import lombok.RequiredArgsConstructor;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.Map;

@Service
@RequiredArgsConstructor(onConstructor = @__(@Autowired))
public class RouteService {

    @NonNull
    private final RouteMapBean routeMapBean;

    public Map<Integer, Route> getAllRoutes() {
        return routeMapBean.getRoutesMap();
    }

    public Route findByRouteId(Integer routeId) {
        return routeMapBean
                .getRoutesMap()
                .getOrDefault(routeId, Route.builder().routeId(routeId).shortName(routeId.toString()).build());
    }
}
